### PR TITLE
Enable end-to-end tests for Oracle and SQLite

### PR DESCRIPTION
**Pull request by oreilles**
Minor tweak: had to replace occurrences of:
```ts
expect(something).toStrictEqual([]);
```
by:
```ts
expect(something).toMatchObject([]);
```
Because tests were failing for SQLite with jest error:
```
Expected: []
Received: serializes to the same string
```

### CODE REVIEW

**Comment by aidenfoxx** [comment]
Just a heads up with Oracle: If your `DB_POOL_MAX` exceeds or matches your `UV_THREADPOOL_SIZE` (default: 4) then the `node-oracledb` library will flood the threadpool and lock up the process. This is really stupid and the `node-oracledb` maintainers don't care to fix it even though no other DB library does this... 😩

More info here: https://github.com/oracle/node-oracledb/blob/main/doc/api.md#-162-connections-threads-and-parallelism

**Comment by Oreilles** [comment]
@aidenfoxx does that mean we should force [`pool.max = 4`](https://knexjs.org/#Installation-pooling) for oracle clients ?

**Comment by aidenfoxx** [comment]
@Oreilles I think the better workaround is to set the `pool.max` to something like:

```
Math.min(
  pool.max,
  Math.max(1, pool.max, process.env.UV_THREADPOOL_MAX - 2)
)
```

That way if sufficient threads are available we use `pool.max`, otherwise we fallback to the thread count minus 2, with a minimum limit of 1.

Always leaves sufficient headroom while still letting the user set their pool limits.

**Comment by Oreilles** [comment]
@rijkvanzanten tests run fine on my machine, it's likely just a memory/cpu issue. 🙂
Finding a way around it.

**Comment by rijkvanzanten** [comment]
@Oreilles Also wondering: Should we run every database test in the same CI process, or try to spin up a copy of the CI for every supported database (using the matrix) so they can run in parallel?

**Comment by Oreilles** [comment]
I'd like to do that, executing the tests in a matrix would solve this performance issue, problem is that it implies seven times checkout + install + build, which even in parallel is long as you have to wait for available instances. I was trying to cache the whole build repo the other day, an back at it rn.

**Comment by Oreilles** [comment]
So, all tests a properly configured and running. :)
They still fail though, because of date issues with Oracle. @aidenfoxx could you help with that ?
```
  ● /items › /:collection GET › "oracle" retrieves artists and events for each entry in artists_events

    insert into "events" ("cost", "created_at", "description", "id", "tags", "time") values (:1, :2, :3, :4, :5, :6) - ORA-01843: not a valid month
```

**Comment by aidenfoxx** [comment]
@Oreilles I've not experienced that one myself. I can look into it tomorrow. 👍

**Comment by rijkvanzanten** [comment]
My guess is that:

https://github.com/directus/directus/blob/aacf92eee6ecb6adc13d0e60e21db981511ff05a/tests/e2e/api/items/many-to-many.test.ts#L29

calls 

https://github.com/directus/directus/blob/aacf92eee6ecb6adc13d0e60e21db981511ff05a/tests/e2e/setup/utils/factories.ts#L143

which is in the wrong value / format / whatever for (node-)oracle?

**Comment by aidenfoxx** [comment]
@rijkvanzanten Yeah, but you'd hope that'd be something knex would handle since I believe it just sends a standard Date object. 🤷

**Comment by Oreilles** [comment]
Why not use Directus itself to seed data ? (and, by extension, also create the schema ?)

**Comment by aidenfoxx** [comment]
I'm with @Oreilles on this one. Feels like it would be way more thorough to run the whole flow as an end user would.

**Comment by rijkvanzanten** [comment]
> Why not use Directus itself to seed data ? (and, by extension, also create the schema ?)

We can! Just gotta be aware at that point that you're using the thing you're testing to prepare the tests, so if there's a bug in that database preparation logic thing, you get a lot of false negatives in the other tests.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,27 +53,29 @@ jobs:
   lint:
     name: Lint
     needs: pre_check
-    uses: directus/directus/.github/workflows/lint.yml@main
+    uses: directus/directus/.github/workflows/lint.yml@add-e2e-tests-vendors
```
Comment (rijkvanzanten) [inline .github/workflows/ci.yml:56]: Should this be changed back to `main`?

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Oreilles) [reply .github/workflows/ci.yml:56]: Yes, same problem as before 👍

**Review by rijkvanzanten — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rijkvanzanten) [reply .github/workflows/ci.yml:56]: Renamed it to `main` on `main` 👍🏻

**Pull request merged**